web/client-api/src/main/java/io/deephaven/web/client/api/widget/JsWidgetExportedObject.java

```diff
@@ -47,24 +46,46 @@ public TypedTicket typedTicket() {
         return typedTicket;
     }
 
-    @JsMethod
+    public boolean isOwner() {
+        return isOwner;
+    }
+
     public Promise<?> fetch() {
-        if (getType().equals(JsVariableType.TABLE)) {
-            return Callbacks.<ExportedTableCreationResponse, Object>grpcUnaryPromise(c -> {
-                connection.tableServiceClient().getExportedTableCreationResponse(ticket.getTicket(),
-                        connection.metadata(),
-                        c::apply);
-            }).then(etcr -> {
-                ClientTableState cts = connection.newStateFromUnsolicitedTable(etcr, "table for widget");
-                JsTable table = new JsTable(connection, cts);
-                // never attempt a reconnect, since we might have a different widget schema entirely
-                table.addEventListener(JsTable.EVENT_DISCONNECT, ignore -> table.close());
-                return Promise.resolve(table);
-            });
-        } else {
-            return this.connection.getObject(
-                    new JsVariableDefinition(ticket.getType(), null, ticket.getTicket().getTicket_asB64(), null));
+        return fetch(false);
+    }
+
+    private void verifyIsOwner() {
+        if (!isOwner) {
+            throw new IllegalStateException("Exported object is no longer the owner of this ticket.");
+        }
+    }
+
+    /**
+     * Takes ownership of this exported object. Returns the ticket for this object.
+     */
+    public TypedTicket takeTicket() {
+        verifyIsOwner();
+        isOwner = false;
+        return typedTicket();
+    }
+
+    /**
+     * Fetches the object from the server.
+     * 
+     * @param takeOwnership Whether to take ownership of the object. Defaults to false.
+     *
+     * @return a promise that resolves to the object
+     */
+    @JsMethod
+    public Promise<?> fetch(@JsOptional Boolean takeOwnership) {
+        if (takeOwnership == null) {
+            takeOwnership = false;
+        }
+        if (takeOwnership) {
+            return this.connection.getObject(takeTicket());
```

Comment:
this means that raced calls to `fetch(true)` and `fetch(false)` are going to intermittently fail - probably need some clearer docs on this and other general usage of ownership.

Reply:
Was debating just having a separate method `take` instead of making it a boolean.
Really you should only be calling one or the other (`true` or `false`) depending on how you're managing the object, I wouldn't expect you'd want to call both.
In any case, I can add a `verifyIsOwner()` here in the `false` case, and throw if it's been called after a `fetch(false)` so it reduces a race that way, but you could still race `fetch(false)` then `fetch(true)` I suppose.